code/modules/overmap/objects/star.dm

```diff
@@ -1,136 +1,169 @@
+	token.bound_height = 96
+	token.bound_width = 96
+	token.pixel_x = -32
+	token.pixel_y = -32
+	. = ..()
 
-/obj/overmap/star
-	name = "Star"
-	desc = "A star."
-	icon = 'whitesands/icons/effects/overmap.dmi'
-	icon_state = "star1"
-	var/star_classes = list(\
-		STARK,
-		STARM,
-		STARL,
-		START,
-		STARY,
-		STARD
-	)
+/datum/overmap/star/giant/yellow
+	token_desc = "Like many other yellow giants, this dying star \"pulsates\" as its brightness fluctuates rhythmically."
+	spectral_type = STAR_G
+	color_vary = 0.25
 
-/obj/overmap/star/Initialize(mapload, new_parent)
+/datum/overmap/star/giant/yellow/alter_token_appearance()
 	. = ..()
-	var/c = "#ffffff"
-	switch(pick(star_classes))
-		if(STARO)
-			c = "#75ffff"
-			desc = "A blue giant."
-		if(STARB)
-			c = "#c0ffff"
-		if(STARG)
-			c = "#ffff00"
-		if(STARK)
-			c = "#ff7f00"
-		if(STARM)
-			c = "#d50000"
-		if(STARL) //Take the L
-			c = "#a31300"
-		if(START)
-			c = "#a60347"
-			desc = "A brown dwarf"
-		if(STARY)
-			c = "#4a3059"
-			desc = "A brown dwarf."
-		if(STARD)
-			c = pick("#75ffff", "#c0ffff", "#ffffff")
-			desc = "A white dwarf."
-	add_atom_colour(c, FIXED_COLOUR_PRIORITY)
-
-/obj/overmap/star/medium
-	icon = 'whitesands/icons/effects/overmap_large.dmi'
-	bound_height = 64
-	bound_width = 64
-	pixel_x = -16
-	pixel_y = -16
-	icon_state = "star2"
-	star_classes = list(\
-		STARB,
-		STARA,
-		STARF,
-		STARG,
-		STARK
-	)
+	// adds a slight, slow flicker
+	// this took me far too long to get working right
+	var/half_duration = rand(7 SECONDS, 15 SECONDS)
+	animate(token, color = token.color, time = half_duration, loop = -1, easing = SINE_EASING, flags = ANIMATION_PARALLEL)
+	animate(color = get_rand_spectral_color(STAR_G - 0.5, 0.125), time = half_duration, easing = SINE_EASING)
 
-/obj/overmap/star/big
-	icon = 'whitesands/icons/effects/overmap_larger.dmi'
-	icon_state = "star3"
-	bound_height = 96
-	bound_width = 96
-	pixel_z = -32
-	pixel_w = -32
-	star_classes = list(\
-		STARO,
-		STARB,
-		STARG,
-		STARM
-	)
+/datum/overmap/star/giant/red
+	token_desc = "Huge and imposing, this red giant has exhausted the hydrogen within its core, and has expanded and brightened as a result. It has begun to die."
+	spectral_type = STAR_M
+
+/*
+		Binary stars
+*/
+
+/datum/overmap/star/binary
+	token_icon_state = "binaryswoosh"
+	token_desc = "Two stars, each locked the other's orbit. These systems form at stellar birth, and may persist long after one or both stars die."
+	color_vary = 0.75
+
+/datum/overmap/star/binary/gen_star_name()
+	return (..() + " AB")
 
-/obj/overmap/star/big/binary
-	icon_state = "binaryswoosh"
-	var/class
-	star_classes = list(\
-		STARK,
-		STARM,
-		STARL,
-		START,
-		STARY,
-		STARD
+/datum/overmap/star/binary/alter_token_appearance()
+	// we don't call the parent proc, since we do colors our own way
+	var/mutable_appearance/star_1
+	var/mutable_appearance/star_2
+	var/list/spectral_types = list(
```

Comment:
```suggestion
	var/static/list/spectral_types = list(
```
I *believe* this will be slightly faster as it won't be re-instantiated every single time it's run